cmd/notation/plugin/install.go

```diff
@@ -0,0 +1,329 @@
+			continue
+		}
+		return installPluginExecutable(ctx, header.Name, pluginName, tarReader, force)
+	}
+	return fmt.Errorf("no valid plugin file was found in %s. Plugin file name must in format notation-{plugin-name}", tarGzPath)
+}
+
+// installPluginExecutable extracts, validates, and installs a plugin file
+func installPluginExecutable(ctx context.Context, fileName string, pluginName string, fileReader io.Reader, force bool) error {
+	// sanity check
+	if runtime.GOOS == "windows" && filepath.Ext(fileName) != ".exe" {
+		return fmt.Errorf("on Windows, plugin executable file %s is missing the '.exe' extension", fileName)
+	}
+	if runtime.GOOS != "windows" && filepath.Ext(fileName) == ".exe" {
+		return fmt.Errorf("on %s, plugin executable file %s cannot have the '.exe' extension", runtime.GOOS, fileName)
+	}
+	// extract to tmp dir
+	tmpDir, err := os.MkdirTemp(".", notationPluginTmpDir)
+	if err != nil {
+		return fmt.Errorf("failed to create notationPluginTmpDir: %w", err)
+	}
+	defer os.RemoveAll(tmpDir)
+	tmpFilePath := filepath.Join(tmpDir, fileName)
+	pluginFile, err := os.OpenFile(tmpFilePath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0700)
+	if err != nil {
+		return err
+	}
+	if _, err := io.Copy(pluginFile, fileReader); err != nil {
+		defer pluginFile.Close()
+		return err
+	}
+	if err := pluginFile.Close(); err != nil {
+		return err
+	}
+	// get plugin metadata
+	pluginMetadata, err := notationplugin.GetPluginMetadata(ctx, pluginName, tmpFilePath)
+	if err != nil {
+		return err
+	}
+	pluginVersion := pluginMetadata.Version
+	// check plugin existence and version
+	var currentPluginVersion string
+	currentPluginMetadata, err := notationplugin.GetPluginMetadataIfExist(ctx, pluginName)
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
+	if err == nil { // plugin already exists
+		currentPluginVersion = currentPluginMetadata.Version
+		if !force {
+			comp, err := notationplugin.ComparePluginVersion(pluginVersion, currentPluginVersion)
+			if err != nil {
+				return err
+			}
+			if comp < 0 {
+				return fmt.Errorf("%s. The installing version %s is lower than the existing plugin version %s.\nIt is not recommended to install an older version. To force the installation, use the \"--force\" option", pluginName, pluginVersion, currentPluginVersion)
+			}
+			if comp == 0 {
+				return fmt.Errorf("%s with version %s already exists", pluginName, currentPluginVersion)
+			}
+		}
+	}
+	// install plugin
+	pluginPath, err := dir.PluginFS().SysPath(pluginName)
+	if err != nil {
+		return err
+	}
+	_, err = osutil.CopyToDir(tmpFilePath, pluginPath)
+	if err != nil {
+		return err
+	}
+	// plugin is always executable
+	pluginFilePath := filepath.Join(pluginPath, fileName)
+	err = os.Chmod(pluginFilePath, 0700)
+	if err != nil {
+		return err
+	}
+	if currentPluginVersion != "" {
+		fmt.Printf("Succussefully installed plugin %s, updated the version from %s to %s\n", pluginName, currentPluginVersion, pluginVersion)
```

Comment:
I think the original message is better, because user used `notation plugin install` command, the message should begin with `Successfully installed...`. @FeynmanZhou